client/galaxy/scripts/mvc/tool/tool-form-base.js

```diff
@@ -66,14 +66,14 @@ define( [ 'utils/utils', 'utils/deferred', 'mvc/ui/ui-misc', 'mvc/form/form-view
             // build request url
             var build_url = '';
             var build_data = {};
-            if ( new_options.job_id ) {
-                build_url = Galaxy.root + 'api/jobs/' + new_options.job_id + '/build_for_rerun';
+            if ( options.job_id ) {
+                build_url = Galaxy.root + 'api/jobs/' + options.job_id + '/build_for_rerun';
```

Comment:
What I have observe is that `options` always contains the `job_id`, but `new_options` only contains changed options. `new_options` only contains the `job_id` when clicking on the rerun icon. So the change in a498390 narrows looking up the `job_id`, but we now also need the `job_id` when having clicked on the rerun icon, followed by switching the tool version (It is not a new option anymore). So maybe it is better to use either `new_options.job_id` or `options.job_id`